libraries/botbuilder-core/src/activityHandler.ts

```diff
@@ -416,38 +400,44 @@ export class ActivityHandler extends ActivityHandlerBase {
     }
 
     /*
-     * Runs all registered _invoke_ handlers and then continues the event emission process.
+     * Provides default behavior for invoke activities.
      * 
      * @param context The context object for the current turn.
      * 
      * @remarks
      * Overwrite this method to support channel-specific behavior across multiple channels.
-     * The default logic is to call any handlers registered via
-     * [onInvoke](xref:botbuilder-core.ActivityHandler.onInvoke),
+     * The default logic is to check for a signIn invoke and handle that
      * and then continue by calling [defaultNextEvent](xref:botbuilder-core.ActivityHandler.defaultNextEvent).
     */
-    protected async onInvokeActivity(context: TurnContext): Promise<void|InvokeResponse> {
-        if(context.activity.name && (context.activity.name === verifyStateOperationName || context.activity.name === tokenExchangeOperationName)) {
-            await this.onSignInInvoke(context);
+    protected async onInvokeActivity(context: TurnContext): Promise<InvokeResponse> {
+        try {
+            if (context.activity.name && (context.activity.name === verifyStateOperationName || context.activity.name === tokenExchangeOperationName)) {
+                await this.onSignInInvoke(context);
+                return { status: 200 };
```

Comment:
> return { status: 200 }; [](start = 15, length = 24)

if there is an oauthprompt in the dialog stack, that would have set an invokeResponse object in the turnstate dictionary for the Invoke_Response_key key, so returning an invokeresponse object here would either throw a key collision exception or worse, override the oauthprompt's invokeresponse with a 200 OK always.
Consider either restoring this back to return a void or checking if the Invoke_Response_key has been set to an invokeresponse. If already set, return that object otherwise return 200

Reply:
the code in ActivityHandlerBase should handle that. Refer to line 70 in that file. The check you are looking for is there.

Reply:
@swagatmishra2007, the scenario you mention is handled by this section of [code](https://github.com/microsoft/botbuilder-js/pull/1872/files#diff-e465969a7fb5affdcc2195df6f6f4ab6R67-R72): 

```ts
// ActivityHandlerBase
case ActivityTypes.Invoke:
    const invokeResponse = await this.onInvokeActivity(context);
    // If onInvokeActivity has already sent an InvokeResponse, do not send another one.
    if (invokeResponse && !context.turnState.get(INVOKE_RESPONSE_KEY)) {
        await context.sendActivity({ value: invokeResponse, type: 'invokeResponse' });
    }
```

Reply:
@swagatmishra2007 if you think this is still a problem, please open an issue, and highlight (or add) a test case that illustrates the problem, thanks.